tests/client.py

```diff
@@ -22,40 +26,41 @@ def query(
         groups: typing.List[str] = ['-'],
         **variables,
     ):
-        response: Response = self.client.post(
+        if not isinstance(username, str):
+            username = username.username
+        response = self.client.post(
             '/graphql',
             json={'query': f""" {query} """, 'variables': variables},
             headers={'groups': json.dumps(groups), 'username': username},
         )
-        return DefaultMunch.fromDict(response.get_json())
+        return DefaultMunch.fromDict(json.loads(response.text))
 
 
 def create_app(db):
-    app = Flask('tests')
+    app = FastAPI(debug=True)
     schema = get_executable_schema()
 
-    @app.route('/', methods=['OPTIONS'])
+    @app.options('/')
     def opt():
         # On GET request serve GraphQL Playground
         # You don't need to provide Playground if you don't want to
         # but keep on mind this will not prohibit clients from
         # exploring your API using desktop GraphQL Playground app.
-        return '<html><body><h1>Hello</h1></body></html>', 200
+        return HTMLResponse('<html><body><h1>Hello</h1></body></html>')
 
-    @app.route('/graphql', methods=['GET'])
+    @app.get('/graphql')
     def graphql_playground():
         # On GET request serve GraphQL Playground
         # You don't need to provide Playground if you don't want to
         # but keep on mind this will not prohibit clients from
         # exploring your API using desktop GraphQL Playground app.
-        return PLAYGROUND_HTML, 200
+        return HTMLResponse(PLAYGROUND_HTML)
 
-    @app.route('/graphql', methods=['POST'])
-    def graphql_server():
+    @app.post('/graphql')
+    async def graphql_server(request: Request):
         # GraphQL queries are always sent as POST
         # Note: Passing the request to the context is optional.
-        # In Flask, the current request is always accessible as flask.request
-        data = request.get_json()
+        data = await request.json()
```

Comment:
Just curious, why is this request async? I might be getting it wrong, but is it just waiting for the request data? Because the graphql_sync call is synchronous no?

Reply:
`.json()` is declared as async thus to get the results we must await it, otherwise we get back a promise.  Thus if you have to call await you must declare the calling method as async.